### PR TITLE
adding other attributes to edit

### LINKED ISSUE #245: [PE-BUG] Edit command to edit all fields in person
As issues #196 #223

### PR DESCRIPTION
**Pull request by SimNee**
adding other attributes to edit Fixes #245

### CODE REVIEW

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
+            }
+        } else if (updatedType.isPatient()) {
+            if (editPersonDescriptor.getOrganExpiryDate().isPresent()) {
+                errorMessage += "There is no organ's expiry date field in patient";
+            }
+        } else if (updatedType.isDonor()) {
+            if (editPersonDescriptor.getPriority().isPresent()) {
+                errorMessage += "There is no priority field in donor\n";
+            }
+            if (editPersonDescriptor.getPriority().isPresent()) {
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: Duplicate code / missing newline?

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
+                errorMessage += "There is no priority field in donor";
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: Donor also does not have doctor in charge

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -173,6 +243,13 @@ public boolean equals(Object other) {
...
         private Type type;
+        private Age age;
+        private Priority priority;
+        private BloodType bloodType;
+        private DoctorInCharge doctorInCharge;
+        private OrganExpiryDate organExpiryDate;
+        private Organ organ;
+        private TissueType tissueType;
 
         public EditPersonDescriptor() {}
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: What is the point of this method?

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -185,13 +262,21 @@ public EditPersonDescriptor(EditPersonDescriptor toCopy) {
             setNric(toCopy.nric);
             setName(toCopy.name);
             setPhone(toCopy.phone);
+            setAge(toCopy.age);
+            setPriority(toCopy.priority);
+            setBloodType(toCopy.bloodType);
+            setDoctorInCharge(toCopy.doctorInCharge);
+            setOrganExpiryDate(toCopy.organExpiryDate);
+            setOrgan(toCopy.organ);
+            setTissueType(toCopy.tissueType);
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: Uh try not to use setter methods in the constructor, its bad practice. Can read more [here](https://stackoverflow.com/questions/16167637/setter-method-in-constructor)

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 5]
```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
+                errorMessage += "There is no organ's expiry date field in doctor\n";
+            }
+            if (editPersonDescriptor.getOrgan().isPresent()) {
+                errorMessage += "There is no organ field in doctor\n";
+            }
+            if (editPersonDescriptor.getDoctorInCharge().isPresent()) {
+                errorMessage += "There is no doctor in charge field in doctor\n";
+            }
+            if (editPersonDescriptor.getPriority().isPresent()) {
+                errorMessage += "There is no priority field in doctor";
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: Missing newline?

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
-            TissueType updatedTissueType = ((Donor) personToEdit).getTissueType();
-            Organ updatedOrgan = ((Donor) personToEdit).getOrgan();
-            OrganExpiryDate updatedOrganExpiryDate = ((Donor) personToEdit).getOrganExpiryDate();
-            Status updatedStatus = ((Donor) personToEdit).getStatus();
-            return new Donor(updatedType, updatedNric, updatedName, updatedPhone, updatedAge,
-                updatedBloodType, updatedTissueType, updatedOrgan, updatedOrganExpiryDate, updatedStatus);
-        } else if (updatedType.isDoctor()) {
+        if (updatedType.isDoctor()) {
+            String errorMessage = getErrorMessage(personToEdit, editPersonDescriptor);
+            if (!errorMessage.equals("")) {
```
Comment (TheRealRavager) [inline src/main/java/organice/logic/commands/EditCommand.java]: May be more intuitive to test .isBlank()

File: src/main/java/organice/logic/commands/EditCommand.java (hunk 3 above)
Comment (WilliamRyank) [reply src/main/java/organice/logic/commands/EditCommand.java]: For testing i think, they use it in `EditPersonDescriptorBuilder` or something like that, so can build an empty person, its from AB3

Comment (TheRealRavager) [reply src/main/java/organice/logic/commands/EditCommand.java]: isnt an empty constructor the default no parameter constructor in java?

File: src/main/java/organice/logic/commands/EditCommand.java [hunk 7]
```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
+            if (editPersonDescriptor.getDoctorInCharge().isPresent()) {
+                errorMessage += "There is no doctor in charge field in doctor\n";
+            }
+            if (editPersonDescriptor.getPriority().isPresent()) {
+                errorMessage += "There is no priority field in doctor\n";
+            }
+        } else if (updatedType.isPatient()) {
+            if (editPersonDescriptor.getOrganExpiryDate().isPresent()) {
+                errorMessage += "There is no organ's expiry date field in patient\n";
+            }
```
Comment (C-likethis123) [inline src/main/java/organice/logic/commands/EditCommand.java]: I suggest the following addition to validate if the edited doctor in charge is valid:

```
if (editPersonDescriptor.getDoctorInCharge().isPresent()) {
        DoctorInCharge updatedDoctorInCharge = editPersonDescriptor.getDoctorInCharge();
       if (model.hasDoctorInCharge(updatedDoctorInCharge)) {
              errorMessage += updatedDoctorInCharge.toString() + " does not exist in ORGANice!";
       }
}
```

**Pull request merged**